presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraIntegrationSmokeTest.java

```diff
@@ -195,6 +203,144 @@ public void testClusteringKeyOnlyPushdown()
+
+        assertEquals(execute("SELECT column_1 FROM cassandra.keyspace_1.table_1").getRowCount(), 1);
+        assertUpdate("DROP TABLE cassandra.keyspace_1.table_1");
+
+        // when an identifier is unquoted the lowercase and uppercase spelling may be used interchangeable
+        session.execute("DROP KEYSPACE keyspace_1");
+    }
+
+    @Test
+    public void testUppercaseNameEscaped()
+            throws Exception
+    {
+        /*
+         * If an identifier is escaped with double quotes it is stored verbatim
+         *
+         * http://docs.datastax.com/en/cql/3.1/cql/cql_reference/ucase-lcase_r.html
+         */
+        session.execute("CREATE KEYSPACE \"KEYSPACE_2\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_2")
+                .build(), new Duration(1, MINUTES));
+
+        session.execute("CREATE TABLE \"KEYSPACE_2\".\"TABLE_2\" (\"COLUMN_2\" bigint PRIMARY KEY)");
+        assertContainsEventually(() -> execute("SHOW TABLES FROM cassandra.keyspace_2"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("table_2")
+                .build(), new Duration(1, MINUTES));
+        assertContains(execute("SHOW COLUMNS FROM cassandra.keyspace_2.table_2"), resultBuilder(getSession(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("column_2", "bigint", "", "")
+                .build());
+
+        // TODO replace with the Presto INSERT INTO once implemented
+        session.execute("INSERT INTO \"KEYSPACE_2\".\"TABLE_2\" (\"COLUMN_2\") VALUES (1)");
+
+        assertEquals(execute("SELECT column_2 FROM cassandra.keyspace_2.table_2").getRowCount(), 1);
+        assertUpdate("DROP TABLE cassandra.keyspace_2.table_2");
+
+        // when an identifier is unquoted the lowercase and uppercase spelling may be used interchangeable
+        session.execute("DROP KEYSPACE \"KEYSPACE_2\"");
+    }
+
+    @Test
+    public void testKeyspaceNameAmbiguity()
+            throws Exception
+    {
+        // Identifiers enclosed in double quotes are stored in Cassandra verbatim. It is possible to create 2 keyspaces with names
+        // that have differences only in letters case.
+        session.execute("CREATE KEYSPACE \"KeYsPaCe_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        session.execute("CREATE KEYSPACE \"kEySpAcE_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+
+        // Although in Presto all the schema and table names are always displayed as lowercase
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_3")
```

Comment:
Why not fail here? Is it so that the rest of Cassandra tables work even if there is a duplicate schema?